astrogrism/simulate/simulate.py

```diff
@@ -96,3 +102,116 @@ def generate_synthetic_spectrum(grism, detector=None, temp_path=gettempdir(), ve
             max_slice = i
             break
     return spectrum[min_slice:max_slice]
+
+
+def disperse_spectrum_on_image(grism_file, wide_field_image, spectrum, detector=None):
+    """
+    Disperses a given spectrum onto an astronomical image
+    along the spectral trace of a specified HST grism
+
+    Parameters
+    ----------
+    grism_file : str or astropy.io.fits.HDUList
+        Grism image (or filepath to one) to construct an Astrogrism GrismObs class
+        TBF: Should be replaced with a Grism identifier only in the future
+
+    wide_field_image : numpy.ndarray or astropy.io.fits.ImageHDU
+        The image or array to disperse the given spectrum atop
+
+    spectrum : specutils.Spectrum1D
+        The spectrum to diperse the on the supplied image
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+    """
+    grismobs = GrismObs(grism_file)
+    if detector:
+        image2grism = grismobs.geometric_transforms[f'CCD{detector}'].get_transform('detector',
+                                                                                    'grism_detector'
+                                                                                    )
+    else:
+        image2grism = grismobs.geometric_transforms.get_transform('detector', 'grism_detector')
+
+    if type(wide_field_image) is ImageHDU:
+        data = wide_field_image.data
+    else:
+        data = wide_field_image
+
+    shape = data.shape
+    simulated_data = np.zeros(shape)
+
+    # Start progressbar
+    bar = progressbar.ProgressBar(maxval=shape[0],
+                                  widgets=[progressbar.Bar('=', '[', ']'),
+                                           ' ',
+                                           progressbar.Percentage()
+                                           ]
+                                  )
+    bar.start()
+
+    # For each pixel in the science image, we need to disperse it's spectrum
+    for x_pixel in range(0, shape[0]):
+        bar.update(x_pixel)
+        for y_pixel in range(0, shape[1]):
+            # Get the flux of the science pixel
+            # We'll need to scale the spectrum to this brightness later
+            data_flux = data[x_pixel][y_pixel]
+            if np.isnan(data_flux) or (data_flux == 0):
+                continue
+
+            # For each Wavelength in the spectrum,
+            # calculate where, in pixels, that wavelength would fall on the detector
+            dispersed_coords = image2grism(x_pixel, y_pixel, spectrum.wavelength, 1)
+            for step in range(0, len(spectrum.wavelength)):
+                spectrum_flux = spectrum.flux[step]
```

Comment:
When I was testing, it looked like this isn't normalized but is still like `40.23 PHOTLAM` or whatever the units are. I think we should normalize the spectrum flux before using it to disperse each data point. Could also get rid of this loop and do it with array handling instead, a la the chunk of code starting [here](https://github.com/duytnguyendtn/astrogrism/pull/1/files#diff-a44ad3e831c16508917324e007e1c88a870570691f40e4410b6fd5782c37b6a1R173), although I think that's pretty much aesthetic not a performance improvement.